From ba360fccc49070e82d85de6811db2e7bcafa653d Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Mon, 6 Dec 2021 23:12:46 +0530
Subject: [PATCH 74/82] arm64: dts: st: add USB2 Host Glue and ehci/ohci nodes
 for stm32mp25

USB2 Host controller Glue logic, along with the standard registers used for
full- and low-speed (OHCI controller) and high-speed (EHCI controller).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 938c3b8bbe793e975041f880bd1b59e13fc18821 arm64: dts: st: add USB2 Host Glue and ehci/ohci nodes for stm32mp25]
Change-Id: I4c573fcd036c6415b7db753539bad0053449501c
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 31 ++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 15b340a3fd7a5..0c998283d3de8 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1793,6 +1793,37 @@ pcie@0,0 {
 					ranges;
 				};
 			};
+
+			usbh: usb@482e0000 {
+				compatible = "st,stm32mp25-usbh";
+				st,syscfg = <&syscfg 0x2420>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0x482e0000 0x482e0000 0x20000>;
+				access-controllers = <&rifsc 63>;
+				status = "disabled";
+
+				usbh_ohci: usb@482e0000 {
+					compatible = "generic-ohci";
+					reg = <0x482e0000 0x1000>;
+					clocks = <&rcc CK_BUS_USB2OHCI>;
+					resets = <&rcc USBH_R>;
+					interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
+					phys = <&usb2_phy1>;
+					phy-names = "usb";
+				};
+
+				usbh_ehci: usb@482f0000 {
+					compatible = "generic-ehci";
+					reg = <0x482f0000 0x1000>;
+					clocks = <&rcc CK_BUS_USB2EHCI>;
+					resets = <&rcc USBH_R>;
+					interrupts = <GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>;
+					companion = <&usbh_ohci>;
+					phys = <&usb2_phy1>;
+					phy-names = "usb";
+				};
+			};
 		};
 
 		bsec: efuse@44000000 {
-- 
2.51.0

